File: Mediation/MoPubExample/Objective-C/VponMopubSample_objC/VponMopubSampleObjC/Mopub/VponMopubBannerViewController.m
//
//  VponMopubBannerViewController.m
//  VponMopubSampleObjC
//
//  Created by EricChien on 2017/6/13.
//  Copyright © 2017年 Soul. All rights reserved.
//

#import "VponMopubBannerViewController.h"

#import <MoPub.h>
#import "MPVponBannerCustomEvent.h"

@interface VponMopubBannerViewController () <MPAdViewDelegate>

@property (weak, nonatomic) IBOutlet UIButton *requestButton;

@property (weak, nonatomic) IBOutlet UIView *loadBannerView;

@property (strong, nonatomic) MPAdView *mpBannerView;

@end

@implementation VponMopubBannerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"MoPub - Banner";
    [self requestButtonDidTouch:self.requestButton];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

#pragma mark - Button Method

- (IBAction)requestButtonDidTouch:(UIButton *)sender {
    
    sender.enabled = NO;
    
    if (self.mpBannerView != nil) {
        [self.mpBannerView removeFromSuperview];
    }
    #warning set ad unit id
    self.mpBannerView = [[MPAdView alloc] initWithAdUnitId:@"782ad2f7565e4241a51045a523f5e12d" size:MOPUB_BANNER_SIZE];
    self.mpBannerView.delegate = self;
    [self.mpBannerView loadAd];
}

#pragma mark - MPAdView Delegate

- (UIViewController *)viewControllerForPresentingModalView {
    return self;
}

- (void)adViewDidLoadAd:(MPAdView *)view {
    NSLog(@"Received banner ad successfully");
    [self.loadBannerView addSubview:view];
    self.requestButton.enabled = YES;
}

- (void)adViewDidFailToLoadAd:(MPAdView *)view {
    NSLog(@"Failed to receive banner");
    self.requestButton.enabled = YES;
}

- (void)willPresentModalViewForAd:(MPAdView *)view {
    NSLog(@"MoPub Banner will present screen");
}

- (void)didDismissModalViewForAd:(MPAdView *)view {
    NSLog(@"MoPub Banner did dismiss");
}

- (void)willLeaveApplicationFromAd:(MPAdView *)view {
    NSLog(@"MoPub Banner will leave publisher application");
}

@end

File: Mediation/MoPubExample/Swift/VponMopubSample_swift/VponMopubSampleSwift/Mopub/VponMopubBannerViewController.swift
//
//  VponMopubBannerViewController.swift
//  VponMopubSampleSwift
//
//  Created by EricChien on 2017/6/13.
//  Copyright © 2017年 Soul. All rights reserved.
//

import UIKit

import MoPub

extension VponMopubBannerViewController: MPAdViewDelegate {
    
    func viewControllerForPresentingModalView() -> UIViewController! {
        return self
    }
    
    func adViewDidLoadAd(_ view: MPAdView!) {
        print("Received banner ad successfully")
        self.loadBannerView.addSubview(view)
        view.translatesAutoresizingMaskIntoConstraints = false
        let constrantX = NSLayoutConstraint.constraints(withVisualFormat: "H:|-0-[view]-0-|", options: NSLayoutConstraint.FormatOptions(rawValue: 0), metrics: [:], views: ["view": view])
        let constrantY = NSLayoutConstraint.constraints(withVisualFormat: "V:|-0-[view]-0-|", options: NSLayoutConstraint.FormatOptions(rawValue: 0), metrics: [:], views: ["view": view])
        self.loadBannerView.addConstraints(constrantX + constrantY)
        view.layoutIfNeeded()
        self.requestButton.isEnabled = true
    }
    
    func adViewDidFail(toLoadAd view: MPAdView!) {
        print("Failed to receive banner")
        self.requestButton.isEnabled = true
    }
    
    func willPresentModalView(forAd view: MPAdView!) {
        print("MoPub Banner will present screen")
    }
    
    func didDismissModalView(forAd view: MPAdView!) {
        print("MoPub Banner did dismiss")
    }
    
    func willLeaveApplication(fromAd view: MPAdView!) {
        print("MoPub Banner will leave publisher application")
    }
}

class VponMopubBannerViewController: UIViewController {
    
    @IBOutlet weak var requestButton: UIButton!
    
    @IBOutlet weak var loadBannerView: UIView!
    
    var mpBannerView: MPAdView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "MoPub - Banner"
        requestButtonDidTouch(requestButton)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    // MARK: Button Method
    
    @IBAction func requestButtonDidTouch(_ sender: UIButton) {
        
        sender.isEnabled = false
        
        if mpBannerView != nil {
            mpBannerView.removeFromSuperview()
        }
        
        mpBannerView = MPAdView(adUnitId: "e036eb60cb694fe7b987f1af41a76eb9")
        mpBannerView.maxAdSize = loadBannerView.frame.size
        mpBannerView.localExtras = ["contentURL":"https://www.vpon.com", "contentData": ["key1": "Mopub", "key2": 1.2, "key3": true]]
        // TODO: set ad unit id
        mpBannerView.delegate = self
        mpBannerView.loadAd()
    }
    
}
